### PR TITLE
chore: deploy mainnet

### LINKED ISSUE #251: (chore) Mainnet deploy: FuelChainState and FuelMessagePortal

### PR DESCRIPTION
**Pull request by DefiCake**
Closes #251

### CODE REVIEW

**Review by K1-R1 — APPROVED** [review]

**Review by PraneshASP — DISMISSED** [review]

**Review by viraj124 — DISMISSED** [review]

File: packages/solidity-contracts/deploy/mainnet/010.gateway_rate_limit_usdc.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+import type { DeployFunction } from 'hardhat-deploy/dist/types';
+import { parseUnits } from 'ethers';
+
+const USDC_ADDRESS = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48';
+const ONE_WEEK_IN_SECONDS = 3600 * 24 * 7;
```
Comment (PraneshASP) [inline packages/solidity-contracts/deploy/mainnet/010.gateway_rate_limit_usdc.ts:6]: can we move these to the constants file? they're being used in multiple scripts.

**Review by PraneshASP — APPROVED** [review]

File: packages/solidity-contracts/deploy/mainnet/006.set_eth_deposit_limit.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import type { HardhatRuntimeEnvironment } from 'hardhat/types';
+import type { DeployFunction } from 'hardhat-deploy/dist/types';
+
+import { FuelMessagePortalV3__factory as FuelMessagePortal } from '../../typechain';
+
+// Rate limit: 10 ETH / week
+const RATE_LIMIT_DURATION = 3600 * 24 * 7;
+
+// Global deposit cap: 100 ETH
+const ETH_DEPOSIT_CAP = parseEther('100');
```
Comment (viraj124) [inline packages/solidity-contracts/deploy/mainnet/006.set_eth_deposit_limit.ts:11]: based on the value in the sheet, this should be `parseEther('7829');`

File: packages/solidity-contracts/deploy/mainnet/006.set_eth_deposit_limit.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    upgrades: { upgradeProxy, erc1967 },
+    deployments: { get, save },
+  } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  const { address } = await get('FuelMessagePortal');
+
+  const constructorArgs = [ETH_DEPOSIT_CAP.toString(), RATE_LIMIT_DURATION];
+
+  await upgradeProxy(address, new FuelMessagePortal(deployer), {
```
Comment (viraj124) [inline packages/solidity-contracts/deploy/mainnet/006.set_eth_deposit_limit.ts:25]: I think we can add the rate limit enable call in this script

File: packages/solidity-contracts/deploy/mainnet/010.gateway_rate_limit_usdc.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  const [deployer] = await ethers.getSigners();
+
+  await deployments.execute(
+    'FuelERC20GatewayV4',
+    { log: true, from: deployer.address },
+    'resetRateLimitAmount',
+    USDC_ADDRESS,
+    // param `limit` must be down/up scaled according to _adjustDepositDecimals
+    // USDC => 6 decimals => it will not have any scaling
+    parseUnits('25000', 6), // 250k USDC
```
Comment (viraj124) [inline packages/solidity-contracts/deploy/mainnet/010.gateway_rate_limit_usdc.ts:19]: this should reflect the sheet, I think like ETH

File: packages/solidity-contracts/deploy/mainnet/010.gateway_rate_limit_usdc.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { parseUnits } from 'ethers';
+
+const USDC_ADDRESS = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48';
+const ONE_WEEK_IN_SECONDS = 3600 * 24 * 7;
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { ethers, deployments } = hre;
+  const [deployer] = await ethers.getSigners();
+
+  await deployments.execute(
```
Comment (viraj124) [inline packages/solidity-contracts/deploy/mainnet/010.gateway_rate_limit_usdc.ts:12]: we can add the enable rate limit call in the same script like ETH

**Review by PraneshASP — APPROVED** [review]

**Pull request merged**